test/tests/commit.js

```diff
@@ -199,6 +199,46 @@ describe("Commit", function() {
     });
   });
 
+  it("can create a signed commit", function() {
+    var expectedCommitId = "b78570159824e6fc161418a626a646933a10ea89";
+
+    // https://github.com/libgit2/libgit2/blob/master/tests/commit/write.c#L362-L384
+    var signature = "-----BEGIN PGP SIGNATURE-----\n" +
+      "Version: GnuPG v1.4.12 (Darwin)\n" +
+      "\n" +
+      "iQIcBAABAgAGBQJQ+FMIAAoJEH+LfPdZDSs1e3EQAJMjhqjWF+WkGLHju7pTw2al\n" +
+      "o6IoMAhv0Z/LHlWhzBd9e7JeCnanRt12bAU7yvYp9+Z+z+dbwqLwDoFp8LVuigl8\n" +
+      "JGLcnwiUW3rSvhjdCp9irdb4+bhKUnKUzSdsR2CK4/hC0N2i/HOvMYX+BRsvqweq\n" +
+      "AsAkA6dAWh+gAfedrBUkCTGhlNYoetjdakWqlGL1TiKAefEZrtA1TpPkGn92vbLq\n" +
+      "SphFRUY9hVn1ZBWrT3hEpvAIcZag3rTOiRVT1X1flj8B2vGCEr3RrcwOIZikpdaW\n" +
+      "who/X3xh/DGbI2RbuxmmJpxxP/8dsVchRJJzBwG+yhwU/iN3MlV2c5D69tls/Dok\n" +
+      "6VbyU4lm/ae0y3yR83D9dUlkycOnmmlBAHKIZ9qUts9X7mWJf0+yy2QxJVpjaTGG\n" +
+      "cmnQKKPeNIhGJk2ENnnnzjEve7L7YJQF6itbx5VCOcsGh3Ocb3YR7DMdWjt7f8pu\n" +
+      "c6j+q1rP7EpE2afUN/geSlp5i3x8aXZPDj67jImbVCE/Q1X9voCtyzGJH7MXR0N9\n" +
+      "ZpRF8yzveRfMH8bwAJjSOGAFF5XkcR/RNY95o+J+QcgBLdX48h+ZdNmUf6jqlu3J\n" +
+      "7KmTXXQcOVpN6dD3CmRFsbjq+x6RHwa8u1iGn+oIkX908r97ckfB/kHKH7ZdXIJc\n" +
+      "cpxtDQQMGYFpXK/71stq\n" +
+      "=ozeK\n" +
+      "-----END PGP SIGNATURE-----";
+
+    var commit_content = "tree 6b79e22d69bf46e289df0345a14ca059dfc9bdf6\n" +
+      "parent 34734e478d6cf50c27c9d69026d93974d052c454\n" +
+      "author Ben Burkert <ben@benburkert.com> 1358451456 -0800\n" +
+      "committer Ben Burkert <ben@benburkert.com> 1358451456 -0800\n" +
+      "\n" +
+      "a simple commit which works\n";
+
+    return NodeGit.Repository.open(reposPath)
+    .then(function(repo) {
+      return repo.createCommitWithSignature(
+        commit_content,
+        signature);
+    })
+    .then(function(commitId) {
+      assert.equal(expectedCommitId, commitId);
```

Comment:
I know you are still working, but I would like to see a test that verifies the signature in the `gpgsig` field of the commit. That will definitely lend credence to fact that we have GPG signing.